docs/source/validation_rules.rst

```diff
@@ -0,0 +1,714 @@
+
+  - Module: is the Python ``re`` module that you want to use. A common one would be search. Refer to Python ``re`` source material to find the most appropriate module to use.
+
+  - Single spaces separate the three strings.
+
+- Example:
+
+  - ``regex search [0-9]{4}\/[0-9]*``
+
+  - The regular expression defined above allows comparison to an expected format of a histological morphology code.
+
+- Default behavior: raises ``error``
+
+- Notes/tips/warnings
+
+  - `regex101.com <https://regex101.com/>`_ is a tool that can be used to build and validate the behavior of your regular expression
+
+  - If the module specified is match for a given attribute's validation rule, regex match validation will be preformed in Google Sheets (but not Excel) real-time during metadata entry.
+
+  - The ``strict_validation parameter`` (in the config.yml file for CLI or in manifest generation REST API calls) sets whether to stop the user from entering incorrect information in a Google Sheets cell (``strict_validation = true``) or simply throws a warning (``strict_validation = false``). Default: ``true``.
+
+  - ``regex`` validation in Google Sheets is different than standard regex validation (for example, it does not support validation of digits). See `this documentation <https://github.com/google/re2/wiki/Syntax>`_ for details on Google regex syntax. It is up to the user/modeler to validate that ``regex match`` is working in their manifests, as intended. This is especially important if the ``strict_validation`` parameter is set to ``True`` as users will be blocked from entering incorrect data. If you are using Google Sheets and do not want to use real-time validation use ``regex search`` instead of ``regex match``.
+
+Type Validation Type
+--------------------
+
+- There are two parameters
+
+  - The first parameter is type and must be one of [ ``float``, ``int``, ```num```, ``str``]
+
+  - The second optional parameter is the msg level and must be one of [ ``error``, ``warning`` ], defaults to ``error``.
+
+- Examples: [ ``str``, ``str error``, ``str warning``]
+
+float
+~~~~~
+
+- Checks that the value is a float.
+
+int
+~~~
+
+- Checks that the value is an integer.
+
+num
+~~~
+
+- Checks that the value is either an integer or float.
+
+str
+~~~
+
+- Checks that the value is a string (not a number).
+
+URL Validation Type
+-------------------
+
+url
+~~~
+
+- Using the ``url`` rule implies the user should add a URL to a free text box as a string. This function will check that the user has provided a usable URL. It will check for any standard URL error and throw an error if one is found. Further additions to this rule can allow for checking that a specific type of URL is added. For example, if the user needs to add a ``http://protocols.io`` URL, ``http://protocols.io`` can be added after url to perform this check. If the provided url does not contain this specific string, an error will be raised.
```

Comment:
I got a bit confused reading this. How about: For example, if the user needs to ensure that the input contains a `http://protocols.io ` URL string, `http://protocols.io` can be added after url to perform this check.